src/components/InterestForm/index.tsx

```diff
@@ -28,89 +30,110 @@ export default function InterestForm({ caseData }: { caseData: CaseListing }) {
   const [reason, setReason] = useState<string>('');
   const [rolesInterested, setRolesInterested] = useState<string>('');
   const [startDate, setStartDate] = useState<string>('');
+  const [submitted, setSubmitted] = useState(false);
+  const [missingInfo, setMissingInfo] = useState(false);
 
   useEffect(() => {
     // Reset form fields when caseData changes
     setReason('');
     setStartDate('');
     setRolesInterested('');
+    setSubmitted(false);
+    setMissingInfo(false);
   }, [caseData]);
 
   const handleInsert = async () => {
     // will improve this in the next sprint
-    if (reason !== '' && startDate !== '' && rolesInterested !== '') {
+    if (startDate === '' || rolesInterested === '') {
+      setMissingInfo(true);
+      return;
+    }
+    if (startDate !== '' && rolesInterested !== '') {
       const newInterest: Interest = {
         // hardcoded values for now
         listing_id: caseData.id,
         listing_type: 'CASE',
-        user_id: '36b8f84d-df4e-4d49-b662-bcde71a8764f',
+        user_id: '515d9b33-9185-4601-9a13-23789bea3ac0',
```

Comment:
next step here is that we want to pull the authenticated user's id from supabase, but let's handle that in the final interest form sprint (!!!!!!!!) which will have that, un-authenticated/un-onboarded states, and the profile match stuff